### ISSUE TITLE
Train new dataset with 77 classes, Result not good

### ISSUE THREAD

**Issue by Dosomecrazy** [HiKapok/SSD.TensorFlow#27]
hi, your work is nice, but i meet some problems about train my own data which has 77 kinds of fruits&veg. i load the pre-trained vgg16 model and train about 86000 steps, followed pics show the result, could you give some suggestions?
![3_result](https://user-images.githubusercontent.com/10924630/43458752-f5179e8c-94fd-11e8-871f-e305ce674709.jpg)
![1_result](https://user-images.githubusercontent.com/10924630/43458776-07fa6cc8-94fe-11e8-9771-f47d343f5a11.jpg)

And i wonder how you solve the loss not converge problem(balancap version), looking forward to your reply, thank you

**Comment by HiKapok** [comment]
@Dosomecrazy There are a lot bugs in balancap's codes, it's hard to list all of them here(maybe I have forgot...). Before we start discuss about the results, could you tell me your input image size? And how to test these images? Did you resize them with aspect ratio unchanged?

**Comment by Dosomecrazy** [comment]
@HiKapok my input image size are 500*500 and 683*683, and not change other default parameter

**Comment by Dosomecrazy** [comment]
@HiKapok can i add your wechat for discussing, my id is 'mjq2290347631', thank you so much !

**Comment by Jiyugaoka** [comment]
@Dosomecrazy  @HiKapok  How do you organize you own dataset? I modified the python script convert_tfrecords.py and dataset_comment.py to make my own dataset with 4 calsses, but the model can't run correctly with Loss = NaN. I have already print the contents of the tfrecords made by my modefied script convert_tfrecords.py and dataset_comment.py, and I can draw the bounding boxes correctly. But I still don't understand why the model can't run. I use tf_debug to debug the model ,but can't understand what it means.

**Comment by HiKapok** [comment]
@Jiyugaoka  [usage example](https://www.tensorflow.org/guide/debugger#finding_nans_and_infs)

**Comment by Dosomecrazy** [comment]
@Jiyugaoka i aslo meet such question when training 13 classes dataset, but it works when training 86 classes dataset, have you solve it ?

**Comment by Jiyugaoka** [comment]
@Dosomecrazy no, I haven't figured out this problem. when I use tf_debug to debug the code , it prints nan values in the bounding boxes coordinates, so I determine to print all the bounding boxes coordinates to find which is nan. hope this works. Maybe we can communicate with wechat, can I add you as a friend on wechat?

**Comment by HiKapok** [comment]
@Jiyugaoka Did you set background class?

**Comment by Jiyugaoka** [comment]
@HiKapok yes, here is the configuration in my modified dataset_common.py
 KITTI_PERSON_LABELS = {
    'none': (0, 'Background'),
    'Car': (1, 'Vehicle'),
    'Van': (2, 'Vehicle'),
    'Truck': (3, 'Vehicle'),
    'person': (4, 'Person'),
}, 
maybe if you have time, you can have a look at my modified convert_tfrecords.py and dataset_common.py, and the annotations of my datasets.

**Comment by HiKapok** [comment]
@Jiyugaoka could you share your dataset_common.py?

**Comment by HiKapok** [comment]
@Jiyugaoka send me your modified codes if you trust me. we can discuss later.

**Comment by Jiyugaoka** [comment]
@HiKapok How can I send you the code ,email or Wechat? Is you school email still used?

**Comment by HiKapok** [comment]
@Jiyugaoka wangchangan@yeah.net

**Comment by Dosomecrazy** [comment]
@HiKapok sorry for disturbing you, i find there losses a cow in result{}.txt when i run eval_ssd.py, so i print the `det_result` and find {'filename': b'', ......}, which should be 000001.jpg

**Comment by HiKapok** [comment]
@Dosomecrazy  did you change dataset when you found this missing error? convert_tfrecords.py will write the image name directly while reading their contents, so it's less possible to produce one record with empty filename but with valid contents.

**Comment by Dosomecrazy** [comment]
@HiKapok my mistake, i confuse the code of tf record producing code. And i aslo meet the loss Nan problem when train 14 classes data (86 class data is running successfully), this is my Nan bug.
![image](https://user-images.githubusercontent.com/10924630/43588298-c46e201a-969e-11e8-911a-caa14ad2f2aa.png)
l double check the xml file bounding box is right, and i tried banlancap version `ssd_losses/cross_entropy_pos/value_1` and 'ssd_losses/localization/value_1' is always zero, only `ssd_losses/cross_entropy_neg/value_1` is deceasing. l don't know how to solve it

**Comment by HiKapok** [comment]
@Dosomecrazy today, we have found there may exist zero width or height (say, ymax equals ymin) in some ground truth bbox, we have solved the NAN loss problem after fixed this. @Jiyugaoka Maybe you have the same problem.

**Comment by Jiyugaoka** [comment]
Now that,we have successfully solved the problem, I want to share my debug experiences here. First and foremost, if you train the Model on you own dataset, make sure you have the correct image file with at least one bbox, and the bbox coordinates must obey the truth that ymin < ymax <= height, xmin < xmax <= width. Then when you encounter a bug,try to use the official debug tool[https://www.tensorflow.org/guide/debugger#debugging_tensorflow_estimators](tf_debuger), it will tell you which value is wrong and where it lies in you source code.Last ,go over the code  again and again until you feel you have understood how it works.
